CONTRIBUTING.md

```diff
@@ -1,3 +1,169 @@
 # Guidelines for contributing
```

Comment:
We are using Title Headers (all Caps except for conjunctions etc.) for all other headers, so let's follow that here as well.